scripts.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Deploys the application to Google Cloud Functions
+deploy_to_functions() {
+  echo 'Deploying to Cloud Functions...';
+  gcloud functions deploy tasks-pizza \
+    --trigger-http \
+    --runtime=nodejs10 \
+    --env-vars-file=.env.yaml
+  echo "Deployed $(gcloud functions describe tasks-pizza --format 'value(httpsTrigger.url)')"
+}
+
+# Deploys the application to Google Cloud Run
+deploy_to_run() {
+  echo 'Deploying to Cloud Run...';
+  gcloud components install beta
+  GCP_PROJECT=$(gcloud config list --format 'value(core.project)' 2>/dev/null)
+  gcloud config set run/region us-central1
+  gcloud builds submit --tag gcr.io/$GCP_PROJECT/tasks-pizza
```

Comment:
If `$GCP_PROJECT` is empty this will error. Use `set -e` to have the script stop instead of attempting (and failing to complete the deployment and extract the URL.

Reply:
We define `$GCP_PROJECT` above.
I think it's reasonable to assume that the gcloud config has a project.

What else should we assume?

Reply:
If this is a first-time setup in a given environment, it's just frustrating. Good practice is to always use `set -eo pipefail` unless there's a reason not to.

Reply:
I've added this best practice.
https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/